IronFrame.Shared/Utilities/JobObject.cs

```diff
@@ -332,13 +361,82 @@ public virtual void TerminateProcessesAndWait(int milliseconds = System.Threadin
         // P/Invoke Helpers
         //
 
-        static Exception Win32LastError(string message, params object[] args)
+        private static Exception Win32LastError(string message, params object[] args)
         {
             var error = Marshal.GetLastWin32Error();
             return new Win32Exception(
-                error, 
-                String.Format(message, args) + 
-                    String.Format(" (Win32 Error Code {0})", error));
+                error,
+                String.Format(message, args) +
+                String.Format(" (Win32 Error Code {0})", error));
+        }
+
+
+        public void SetJobCpuLimit(int weight)
+        {
+            if (weight < 1 || weight > 9)
+            {
+                throw new ArgumentOutOfRangeException("weight", weight, "CPU Limit must be between 1 and 9");
+            }
+            int length = Marshal.SizeOf(typeof(NativeMethods.JobObjectCpuRateControlInformation));
+            IntPtr cpuRatePtr = IntPtr.Zero;
+
+            try
+            {
+                var cpuRate = new NativeMethods.JobObjectCpuRateControlInformation
+                {
+                    ControlFlags = (UInt32)(NativeMethods.JobObjectCpuRateControl.Enable | NativeMethods.JobObjectCpuRateControl.WeightBased),
+                    Weight = (UInt32)weight,
+                };
+
+                cpuRatePtr = Marshal.AllocHGlobal(length);
+
+                Marshal.StructureToPtr(cpuRate, cpuRatePtr, false);
+
+                if (
+                    !NativeMethods.SetInformationJobObject(handle,
+                        NativeMethods.JobObjectInfoClass.CpuRateControlInformation, cpuRatePtr, length))
+                {
+                    throw new Win32Exception(Marshal.GetLastWin32Error());
+                }
+            }
+            finally
+            {
+                if (cpuRatePtr != IntPtr.Zero)
+                    Marshal.FreeHGlobal(cpuRatePtr);
+            }
+        }
+
+        public virtual int GetJobCpuLimit()
+        {
+            int infoSize = Marshal.SizeOf(typeof(NativeMethods.JobObjectCpuRateControlInformation));
+            IntPtr infoPtr = IntPtr.Zero;
+            try
+            {
+                infoPtr = Marshal.AllocHGlobal(infoSize);
```

Comment:
You should be able to use `SafeAllocation` here too.

Since the compiler can't infer the type, you'll need to be explicit: `SafeAllocation<NativeMethods.JobObjectCpuRateControlInformation>`